pdns/lwres.cc

```diff
@@ -341,16 +341,27 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
   }
   else {
     try {
-      const int timeout = g_networkTimeoutMsec / 1000; // XXX tcpiohandler's unit is seconds
+      const struct timeval timeout{ g_networkTimeoutMsec / 1000, g_networkTimeoutMsec % 1000 * 1000};
 
       Socket s(ip.sin4.sin_family, SOCK_STREAM);
       s.setNonBlocking();
       localip = pdns::getQueryLocalAddress(ip.sin4.sin_family, 0);
       s.bind(localip);
 
       std::shared_ptr<TLSCtx> tlsCtx{nullptr};
-      TCPIOHandler handler("", s.releaseHandle(), timeout, tlsCtx, now->tv_sec);
-      IOState state = handler.tryConnect(SyncRes::s_tcp_fast_open_connect, ip);
+      if (SyncRes::s_dot_to_port_853 && ip.getPort() == 853) {
+        TLSContextParameters tlsParams;
+        tlsParams.d_provider = "openssl";
```

Comment:
Just a note but if you ever want to support GnuTLS, I have fixes for client-side TLS in #10338.